dashboard/public/pixelation/pixelation.js

```diff
@@ -0,0 +1,365 @@
+/**
+ * Pixelation widget for visualizing image encoding.
+ *
+ * Original code written by Baker Franke.
+ */
+
+var MAX_SIZE = 400;
+
+var pixel_format = document.querySelector('#pixel_format');
+var pixel_data = document.querySelector("#pixel_data");
+var canvas = document.querySelector("#canvas");
+var ctx = canvas.getContext("2d");
+
+function isHex() {
+  return "hex" == document.querySelector('input[name="binHex"]:checked').value;
+}
+
+function drawGraph() {
+  ctx.fillStyle = "#000";
+  ctx.fillRect(0, 0, MAX_SIZE, MAX_SIZE);
+
+  var binCode = "";
+  var hexMode = isHex();
+
+  // If the hex radio button is currently selected.
+  if (hexMode) {
+    // Then we need to get the binary representation for this function to work.
+    // The text area should preserve line breaks, spaces, and hex digits.
+    pixel_data.value = pixel_data.value.replace(/[^0-9A-F \n]/gi, "");
+
+    // Binary code needs to have everything stripped and converted.
+    binCode = hexToBinPvt(pixel_data.value.replace(/[^0-9A-F]/gi, ""));
+  } else {
+    // Otherwise take the straight binary from the text input.
+    // The text area should preserve line breaks, spaces, 0s, and 1s.
+    pixel_data.value = pixel_data.value.replace(/[^01 \n]/gi, "");
+
+    // Binary code needs to have everything stripped except for 0s and 1s.
+    binCode = pixel_data.value.replace(/[^01]/gi, "");
+  }
+
+  var w, h, bitsPerPix = 1;
+  if (options.version == '1') {
+    w = document.getElementById("width").value;
+    h = document.getElementById("height").value;
+  } else {
+    // Read width, height out of the bit string (where width is given in byte 0, height in byte 1).
+    w = binToInt(readByte(binCode, 0));
+    h = binToInt(readByte(binCode, 1));
+    document.getElementById("width").value = document.getElementById("widthRange").value = w;
+    document.getElementById("height").value = document.getElementById("heightRange").value = h;
```

Comment:
Personal preference: Extracting all the `document.getElementById` calls to assignments so you only call it once per element.

```javascript
var widthInput = document.getElementById("width");
if (options.version == '1') {
  w = widthInput.value;
} else {
  w = binToInt(readByte(binCode, 0));
  widthInput.value = w;
  //...
}
```